web-module/src/main/webapp/resources/structEditJS/autocomplete.js

```diff
@@ -51,12 +55,15 @@
     const language = ((window.celMessages && window.celMessages.celmeta)
         ? window.celMessages.celmeta.language : '')
         || 'de';
-    document.querySelectorAll('.structAutocomplete').forEach(selectElem => {
-      console.log('initAutocomplete:', selectElem, language)
-      const type = selectElem.dataset.autocompleteType || '';
-      const cellRef = selectElem.dataset.cellRef || '';
-      $j(selectElem).select2(buildSelect2Config(type, language, cellRef))
-      $j(selectElem).on('select2:unselect', clearSelectOptions);
+    document.querySelectorAll('.structAutocomplete:not(.initialised)').forEach(selectElem => {
+      try {
+        $j(selectElem).select2(buildSelect2Config(selectElem, language))
+        $j(selectElem).on('select2:unselect', clearSelectOptions);
+        selectElem.classList.add('initialised');
```

Comment:
eventhough js is single threaded it is better understantable that this is the invariant preventing double initialization if you do it as first command. Please move before select2 init on line 60

Reply:
it makes a difference in case `select2` fails. yet if that happens, it doesn't make sense to retry it after every content changed.

Reply:
for this meaning you can still add a second css class (e.g. "initSucessfull") to not only inticate that it was initialized, but it was successfully initialized.